src/hdtNIFValidator.cpp

```
@@ -0,0 +1,518 @@
+		auto fileSize = file.tellg();
+		if (fileSize <= 0 || fileSize > 256 * 1024 * 1024) {
+			return result;
+		}
+
+		std::vector<uint8_t> data(static_cast<size_t>(fileSize));
+		file.seekg(0);
+		file.read((char*)data.data(), fileSize);
+		file.close();
+
+		// Find the null-terminated ASCII header
+		size_t headerEnd = 0;
```

Clang (14.0.6)

[warning] variable 'headerEnd' is not initialized

(cppcoreguidelines-init-variables)